### ISSUE TITLE
Ternary expressions with null fail

### ISSUE THREAD

**Issue by jameslyon** [typetools/checker-framework#829]
The following program is no longer considered valid under v2.1.0:

```java
import javax.annotation.Nullable;

public class Checker {
  @Nullable
  public static Double getDouble(boolean flag) {
    return flag ? null : 1.0;
  }
}
```

This works under v2.0.0. The error message is:

```
Checker.java:8: error: [type.invalid] [@Initialized, @Nullable] may not be applied to the type "@Initialized @Nullable double"
    return flag ? null : 1.0;
```

**Comment by smillst** [comment]
Thanks for reporting this issue.  I've added a test case to checker/tests/nullness/Issue829.java.  And I've opened a PR (#830) with a fix.
